site/src/pages/WorkspacePage/WorkspaceActions/WorkspaceActions.tsx

```diff
@@ -57,81 +57,72 @@ export const WorkspaceActions: FC<WorkspaceActionsProps> = ({
   handleUpdate,
   handleCancel,
   handleSettings,
+  handleRetry,
+  handleRetryDebug,
   handleChangeVersion,
-  handleDormantActivate: handleDormantActivate,
+  handleDormantActivate,
   isUpdating,
   isRestarting,
   canChangeVersions,
+  canRetryDebug,
 }) => {
-  const {
-    canCancel,
-    canAcceptJobs,
-    actions: actionsByStatus,
-  } = actionsByWorkspaceStatus(
-    workspace,
-    workspace.latest_build.status,
-    canChangeVersions,
-  );
-  const canBeUpdated = workspace.outdated && canAcceptJobs;
   const { duplicateWorkspace, isDuplicationReady } =
     useWorkspaceDuplication(workspace);
 
-  // A mapping of button type to the corresponding React component
-  const buttonMapping: ButtonMapping = {
-    [ButtonTypesEnum.update]: <UpdateButton handleAction={handleUpdate} />,
-    [ButtonTypesEnum.updating]: (
-      <UpdateButton loading handleAction={handleUpdate} />
-    ),
-    [ButtonTypesEnum.start]: (
-      <StartButton workspace={workspace} handleAction={handleStart} />
-    ),
-    [ButtonTypesEnum.starting]: (
+  // A mapping of button type to their corresponding React components
+  const buttonMapping: Record<ButtonType, ReactNode> = {
+    update: <UpdateButton handleAction={handleUpdate} />,
```

Comment:
Lot of diffs, but only two major things changed:
- Enums got factored out, cleaning up how the properties were defined
- `retry` and `retryDebug` properties got added

Reply:
I think that enum was my complication; thanks for fixing :)